CONTRIBUTING.md

```diff
@@ -121,11 +121,49 @@ TODO
         into fixing and (2) otherwise isn't being prioritized are likely to
         be closed.
 
-### What to expect when you submit a PR
-
-TODO: It is strongly recommended that you validate your contributions before
-you make significant efforts…
-
-The "Assigned" field on a PR indicates who has taken responsibility
-for reviewing the PR, not who is responsible for the content of the
-PR.
+### Pull requests
+
+The "Assigned" field on a pull request indicates who has taken
+responsibility for shepherding it through the review process, not who
+is responsible for authoring it. The assignee is usually the reviewer,
+but they can also delegate the review to someone else. The intent of
+assignment is simply to ensure that pull requests don't get neglected.
+
+A draft pull request is taken to be not yet ready for review. Marking
+drafts as such helps the maintainers triage review work.
+
+When one pull request builds on another, please put "Depends on #NNNN"
+towards the top of its description. This helps maintainers notice that
+they shouldn't merge it until its ancestor has been approved. Don't
+use the "draft pull request" status to indicate dependency.
+
+If a pull request contains multiple commits, please indicate whether
+they need to be squashed into a single commit before they're merged,
+or if they're ready to rebase onto `trunk` as they stand. In the
+latter case, please ensure that each commit passes all CI tests, so
+that we can continue to bisect along `trunk` to isolate bugs.
```

Comment:
Could we put both of these in the PR template? Feel free to make a separate PR if needed.